server/src/internalClusterTest/java/org/elasticsearch/snapshots/RepositoriesIT.java

```diff
@@ -225,4 +229,106 @@ public void testRepositoryVerification() {
             assertThat(ExceptionsHelper.stackTrace(ex), containsString("is not shared"));
         }
     }
+
+    public void testSnapshotDeleteWithRepositoryOutage() throws Exception {
+        disableRepoConsistencyCheck("This test expects residual files in repository");
+        Client client = client();
+        Path repo = randomRepoPath();
+        logger.info("-->  creating repository at {}", repo.toAbsolutePath());
+        assertAcked(client.admin().cluster().preparePutRepository("test-repo")
+            .setType("mock").setSettings(Settings.builder()
+                .put("location", repo)
+                .put("compress", false)
+                .put("chunk_size", randomIntBetween(100, 1000), ByteSizeUnit.BYTES)));
+
+        logger.info("--> creating index-1 and ingest data");
+        createIndex("test-idx-1");
+        ensureGreen();
+        for (int j = 0; j < 10; j++) {
+            indexDoc("test-idx-1", Integer.toString( 10 + j), "foo", "bar" +  10 + j);
+        }
+        refresh();
+
+        // Make repository to throw exception when trying to delete stale indices
+        String masterNode = internalCluster().getMasterName();
+        ((MockRepository)internalCluster().getInstance(RepositoriesService.class, masterNode).repository("test-repo"))
+            .setThrowExceptionWhileDelete(true);
+
+        logger.info("--> creating snapshot");
+        CreateSnapshotResponse createSnapshotResponse = client.admin().cluster().prepareCreateSnapshot("test-repo", "test-snap-1")
+            .setWaitForCompletion(true).get();
+        assertThat(createSnapshotResponse.getSnapshotInfo().successfulShards(), greaterThan(0));
+        assertThat(createSnapshotResponse.getSnapshotInfo().successfulShards(),
+            equalTo(createSnapshotResponse.getSnapshotInfo().totalShards()));
+
+        logger.info("--> delete the snapshot");
+        client.admin().cluster().prepareDeleteSnapshot("test-repo", "test-snap-1").get();
+
+        logger.info("--> done");
+    }
+
+    public void testResidualStaleIndicesAreDeletedByConsecutiveDelete() throws Exception {
+        Client client = client();
+        Path repo = randomRepoPath();
+        logger.info("-->  creating repository at {}", repo.toAbsolutePath());
```

Comment:
We have a shortcut here now as well `org.elasticsearch.snapshots.AbstractSnapshotIntegTestCase#createRepository(java.lang.String, java.lang.String, java.nio.file.Path)` :)

Reply:
Updated the code